de.dentrassi.pm.aspect.mvnosgi/src/de/dentrassi/pm/aspect/mvnosgi/internal/VirtualizerImpl.java

```diff
@@ -43,15 +42,41 @@ public void virtualize ( final Context context )
 
     private void processVirtualize ( final Context context ) throws Exception
     {
-        final ArtifactInformation art = context.getArtifactInformation ();
+        // The group ID will be set using the provided channel meta data or the default one.
+        String groupId = null;
+
+        // Extract some informations (e.g. group ID) from the provided channel meta data.
+        final Map<MetaKey, String> channelMetaData = context.getProvidedChannelMetaData ();
+        for ( final Entry<MetaKey, String> entry : channelMetaData.entrySet () )
```

Comment:
You already stated that this is the quick memory access, so both solutions should be fine, but let's have a look at this.
But let's have a look at, it is good practice to do some brain jogging.


If every provided channel meta data is used (why we should provide a meta data and not use them), we get O(n * log(n)) vs. O(n).

We could say, the code is written once and independent of n (so the meta data used is constant at runtime, you are right).

The documentation of "getProvidedChannelMetaData" states:
This call only returns the provided channel meta data, since the extracted (aggregated) channel meta data will only be available after the virtualizer or generator have run an cannot be used therefore.

So, if the provided meta data is that of the correspondent web UI, the number of entries in the map is also fixed at runtime.

I am missing something?